src/main/java/org/fcrepo/importexport/importer/Importer.java

```diff
@@ -337,25 +349,71 @@ private void importFile(final File f) {
         final String sourceRelativePath =
                 config.getBaseDirectory().toPath().relativize(f.toPath()).toString();
         final String filePath = f.getPath();
+
+        //ignore header files
+        if (filePath.endsWith(".headers")) {
+            // this could be hidden files created by the OS
+            logger.debug("Skipping .headers files ({}).", sourceRelativePath);
+            return;
+
+        }
+
+        //parse headers from headers file
+        final Map<String,List<String>> headers;
+
+        try {
+            headers = parseHeaders(getHeadersFile(f));
+        } catch (IOException ex) {
+            importLogger.error(String.format("Error reading/parsing headers file for %1$s - Message: %2$s",
+                f.getAbsolutePath(), ex.getMessage()), ex);
+            throw new RuntimeException(
+                "Error reading or parsing headers file for" + f.getAbsolutePath() + ": " + ex.toString(), ex);
+        }
+
+        //always skip timemaps since they are derived from the mementos they contain.
+        if (isTimeMap(headers)) {
+            logger.debug("Skipping {} :  TimeMaps are never imported.", sourceRelativePath);
+            return;
+        }
+
+        final boolean isMemento = isMemento(headers);
+
+        //skip versions when include versions flag is false.
+        if (!config.includeVersions() && isMemento) {
+            logger.debug("Skipping {}: Versions import disabled.", sourceRelativePath);
+            return;
+        }
+
         if (filePath.endsWith(BINARY_EXTENSION) || filePath.endsWith(EXTERNAL_RESOURCE_EXTENSION)) {
             // ... this is only expected to happen when binaries and metadata are written to the same directory...
-            if (config.isIncludeBinaries()) {
-                logger.debug("Skipping binary {}: it will be imported when its metadata is imported.",
+
+            if (!isMemento) {
+                if (config.isIncludeBinaries()) {
+                    logger.debug("Skipping binary {}: it will be imported when its metadata is imported.",
                         sourceRelativePath);
+                } else {
+                    logger.debug("Skipping binary {}", sourceRelativePath);
+                }
+                return;
             } else {
-                logger.debug("Skipping binary {}", sourceRelativePath);
+                //continue processing
```

Comment:
extremely minor -- should this whole else just be a comment?  ie:  `// else continue processing`  versus having the `else {}`?

Reply:
good point.